tests/e2e/LicenseCheckerTest.php

```diff
@@ -0,0 +1,172 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\E2E\Lendable\ComposerLicenseChecker;
+
+use Lendable\ComposerLicenseChecker\LicenseChecker;
+use Lendable\ComposerLicenseChecker\LicenseConfiguration;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Console\Tester\CommandTester;
+use Tests\Support\Lendable\ComposerLicenseChecker\LicenseConfigurationFileBuilder;
+
+final class LicenseCheckerTest extends TestCase
+{
+    private CommandTester $commandTester;
+
+    private static string $previousCwd;
+
+    public static function setUpBeforeClass(): void
+    {
+        self::$previousCwd = \getcwd();
+        \chdir(__DIR__.DIRECTORY_SEPARATOR.'..'.DIRECTORY_SEPARATOR.'data');
+    }
+
+    public static function tearDownAfterClass(): void
+    {
+        \chdir(self::$previousCwd);
+    }
+
+    protected function setUp(): void
+    {
+        $command = new LicenseChecker();
+        $command->setAutoExit(false);
+
+        $this->commandTester = new CommandTester($command);
+    }
+
+    public function test_failure_with_non_existent_allowed_licenses_file(): void
+    {
+        $exitCode = $this->commandTester->execute(['--allow-file' => 'dont_have.pls']);
+        $output = $this->getOutputLines();
+
+        self::assertSame(1, $exitCode);
+        self::assertCount(4, $output);
+        self::assertSame('[ERROR] File "dont_have.pls" could not be read.', $output[3]);
```

Comment:
Probably a bit fragile, why are there 4 lines, why is that message specifically at index 3 etc?

Reply:
(probably all that we care about is that it failed and for the expected reason, perhaps we can get that confidence with some less fragile assertions)

Reply:
```
.array(4) {
  [0]=>
  string(24) "Composer License Checker"
  [1]=>
  string(24) "========================"
  [2]=>
  string(0) ""
  [3]=>
  string(47) "[ERROR] File "dont_have.pls" could not be read."
}
```

:shrug:

Reply:
I mean, you could array filter over and just check the message you want it somewhere in the output, plus the correct exit code. Why do I care that it's at index 3?

Reply:
We care that the output is an expected output, nothing more, nothing less. We don't care about the header (title?) part, which is always first 3 lines... but maybe we should also care about that? :see_no_evil:

Reply:
nit: maybe just compare it against a plain text expected outcome fixture file?